code/game/machinery/doors/firedoor.dm

```diff
@@ -78,17 +82,153 @@
+			return
+		number_of_people_trying_to_open++
+
+		//dont want them to be moved by pressure differences when theyre holding onto the firelock with all their strength
+		user.move_resist = MOVE_FORCE_VERY_STRONG
+
+		//starting stats in order to write logs that provide enough data to make kickass fucking graphs later
+		var/starting_health = user.health
+		var/starting_time = REALTIMEOFDAY
+
+		var/datum/gas_mixture/users_air = user.return_air()
+		var/user_loc_starting_temperature = users_air.temperature
+
+		///we dont want people with only one hand to have the same visible_message as people with two hands
+		var/hand_string = "hands"
+		if(iscarbon(user))
+			var/mob/living/carbon/user_as_carbon = user
+			var/obj/item/bodypart/right_arm = user_as_carbon.get_bodypart(BODY_ZONE_R_ARM)
+			var/obj/item/bodypart/left_arm = user_as_carbon.get_bodypart(BODY_ZONE_L_ARM)
+
+			if(!right_arm || right_arm.bodypart_disabled)//this is assuming that you have at least one hand since... this proc is called attack_hand
+				hand_string = "hand"
+			if(!left_arm || left_arm.bodypart_disabled)
+				hand_string = "hand"
+
+		//we want to acknowledge that several people are trying to open the same firelock to encourage them to do it
+		//also to signal to people doing it by themselves that its faster with help
+		if(number_of_people_trying_to_open == 1)
+			user.visible_message("<span class='notice'>[user] tries to open \the [src] with their [hand_string], struggling greatly to open the heavy door by themselves.</span>", \
+				"<span class='notice'> You try with all your strength to pry open \the [src] with your [hand_string], barely moving it!</span>")
+		else if (number_of_people_trying_to_open == 2)
+			user.visible_message("<span class='notice'>[user.name] joins another in trying to open \the [src] with their [hand_string].</span>", \
+				"<span class='notice'> You join another in trying to pry open \the [src] with your [hand_string]!</span>")
+		else
+			user.visible_message("<span class='notice'>[user] joins the others in trying to open \the [src] with their [hand_string]!</span>", \
+				"<span class='notice'> You join the others in trying to pry open \the [src] with your [hand_string]!</span>")
+
+		//burn arms in the do_after callback so that its applied continuosly instead of at the start and/or end
+		var/datum/callback/burning_callback = CALLBACK(src, .proc/burn_arms, user)
+
+		//this callback adjusts the timer of the do_after_dynamic for each user if the number of people trying to open the door changes
+		var/datum/callback/timer_callback = CALLBACK(src, .proc/adjust_do_after_timer)
+
+		///players can team up to open it faster, but only up to a point. 20 seconds -> 13.333 -> 8.88
+		var/true_opening_time = TOOLLESS_OPEN_DURATION_SOLO / max((min(3, number_of_people_trying_to_open) * 0.75), 1)
+
+		if(do_after_dynamic(user, true_opening_time, src, extra_checks = burning_callback, dynamic_timer_change = timer_callback))
```

Comment:
‼️ `extra_checks` should NOT be performing side effects. `extra_checks` should be a 100% pure check. Its interval is not defined anywhere. Have the side effects be on a separate piece, such as on a process that is started when somebody starts to open with their hands.